src/devices/RelayBoard/RelayBoard.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Spi;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+
+using RelayBoard;
+
+namespace Iot.Device.RelayBoard
+{
+    /// <summary>
+    /// A board containing some <see cref="Relay"/> devices.
+    /// </summary>
+    /// <example>
+    /// Create a relay board, with a relay:
+    /// <code>
+    /// RelayBoard board = new RelayBoard();
+    /// board.CreateRelay(pin: 1);
+    /// </code>
+    /// </example>
+    public class RelayBoard : IDisposable, IEnumerable<Relay>
+    {
+        private List<Relay> _relays = new List<Relay>();
+
+        private bool _shouldDispose;
+        private GpioController _controller;
+
+        /// <summary>
+        /// Get's the type of relay board.
+        /// </summary>
+        public RelayType Type { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelayBoard"/> class.
+        /// </summary>
+        public RelayBoard(RelayType relayType = RelayType.NormallyClosed, GpioController gpioController = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = gpioController == null || shouldDispose;
+            _controller = gpioController ?? new GpioController(pinNumberingScheme);
+            Type = relayType;
+        }
+
+        /// <summary>
+        /// Create and initialize a relay on this board.
+        /// </summary>
+        /// <param name="pin">Pin to use.</param>
+        public void CreateRelay(int pin)
+        {
+            _relays.Add(new Relay(pin, _controller, Type));
+        }
+
+        /// <summary>
+        /// Create and initialize relays on the specified pins.
+        /// </summary>
+        /// <param name="pins">List of pins to create relays on.</param>
+        public void CreateRelays(params int[] pins)
```

Comment:
I feel like this should be a constructor parameter and we should not have this API directly - you should know what relays you have up front - I do not think it will be a common scenario that someone will plug & play relays, and if we find in the future that it is then we should consider doing it

Reply:
My thought was if you are dynamically configuring a device - eg the Pi has a web portal, where you can add or remove components/relays/etc. or dynamically set what each pin does through said UI. However, we can change this if you want.

Reply:
@krwq I could see this being very useful in an initializer function or stage

If this is a constructor argument, in a DI scenario, you'll need to provide a custom resolver func for the board which pulls your pin configuration, and passes the pins as appropriate. Not a lot of overhead, but just enough to muddy up type registration.

I'm using IHostedService as an example, which provides a StartAsync function - if registered correctly, this can serve as an initializer before your web server or any other hosted services are started and begin to . Other implementations will use different means.

In this initializer stage, you can configure your hardware - open the pins on gpio controller to use the desired mode, ensure connectivity between devices, etc. And in this case create the appropriate relays from configuration - all before it's actually used by application code.

I get this is probably a matter of semantics, but if we're voting, I'd like to see this stay.